staging/src/k8s.io/apiserver/pkg/storage/etcd/api_object_versioner.go

```diff
@@ -81,6 +82,35 @@ func (a APIObjectVersioner) ObjectResourceVersion(obj runtime.Object) (uint64, e
 	return strconv.ParseUint(version, 10, 64)
 }
 
+// ParseWatchResourceVersion takes a resource version argument and converts it to
+// the etcd version we should pass to helper.Watch(). Because resourceVersion is
+// an opaque value, the default watch behavior for non-zero watch is to watch
+// the next value (if you pass "1", you will see updates from "2" onwards).
+func (a APIObjectVersioner) ParseWatchResourceVersion(resourceVersion string) (uint64, error) {
```

Comment:
I'm confused why we have both ParseWatchResourceVersion and ParseListResourceVersion... they differ only in the type of error they return. That means they only differ in error cases, not in what they return that should be passed to helper.Watch(), right? It seems really odd that a resource version would be parsed differently depending on the intended use.

Reply:
I believe it is the input that differs. I agree the error handling is weird, but I'd like that fixed in another PR; this one is not changing any behavior.

Reply:
I made the errors consistent, but if it causes any test failures I'll change it back.

Reply:
> I believe it is the input that differs. I agree the error handling is weird, but I'd like that fixed in another PR; this one is not changing any behavior.

if the transformation is consistent (which it is), then it will work for any input. would like to see these collapsed to a single `ParseResourceVersion`, but a follow up is fine.